tests/core_tests/service_nodes.cpp

```diff
@@ -794,6 +789,9 @@ bool test_swarms_basic::generate(std::vector<test_event_entry>& events)
 
   DO_CALLBACK(events, "test_after_deregisters");
 
+  /// test (implicitly) that deregistered nodes do not receive rewards
+  gen.rewind_blocks_n(5);
```

Comment:
How does this work? All nodes have expired/deregistered at this point?

Reply:
We deregister a few above (with `build_deregister`). This affects the order in which rewards are distributed and would result in a block verification error if the nodes weren't deleted.